src/NuGet.Packaging/PackagesConfigReader.cs

```diff
@@ -122,11 +123,21 @@ public NuGetVersion GetMinClientVersion()
         }
 
         /// <summary>
-        /// Reads all package node entries in the config
+        /// Reads all package node entries in the config.
+        /// If duplicate package ids exist an exception will be thrown.
         /// </summary>
-        /// <returns></returns>
         public IEnumerable<PackageReference> GetPackages()
         {
+            return GetPackages(allowDuplicatePackageIds: false);
+        }
+
+        /// <summary>
+        /// Reads all package node entries in the config.
+        /// </summary>
+        /// <param name="allowDuplicatePackageIds">If True validation will be performed to ensure that 
+        /// only one entry exists for each unique package id.</param>
+        public IEnumerable<PackageReference> GetPackages(bool allowDuplicatePackageIds)
```

Comment:
No it will always be true

Sent from my Windows Phone
________________________________
From: daazose<mailto:notifications@github.com>
Sent: ‎9/‎29/‎2015 12:48 PM
To: NuGet/NuGet3<mailto:NuGet3@noreply.github.com>
Cc: Yishai Galatzer<mailto:yigalatz@microsoft.com>
Subject: Re: [NuGet3] PackagesConfigReader support for duplicate package id entries (part 1) (#134)


In src/NuGet.Packaging/PackagesConfigReader.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet3%2fpull%2f134%23discussion_r40719555&data=01%7c01%7cyigalatz%40microsoft.com%7cc37d5f86c0e44a2520b608d2c906e0a2%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=EiZQBge9Psgp2tmB%2bFFqmlu%2fmroaDHuLatKrpoJ9%2fzU%3d>:

>          public IEnumerable<PackageReference> GetPackages()
>          {
> +            return GetPackages(allowDuplicatePackageIds: false);
> +        }
> +
> +        /// <summary>
> +        /// Reads all package node entries in the config.
> +        /// </summary>
> +        /// <param name="allowDuplicatePackageIds">If True validation will be performed to ensure that
> +        /// only one entry exists for each unique package id.</param>
> +        public IEnumerable<PackageReference> GetPackages(bool allowDuplicatePackageIds)


@yishaigalatzer<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fyishaigalatzer&data=01%7c01%7cyigalatz%40microsoft.com%7cc37d5f86c0e44a2520b608d2c906e0a2%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=GmfyKLsI6eap8kADhfdSrCWvrZmeZJGGMuxakl7kE%2fI%3d> are there plans to expose this as an option to the restore command?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet3%2fpull%2f134%2ffiles%23r40719555&data=01%7c01%7cyigalatz%40microsoft.com%7cc37d5f86c0e44a2520b608d2c906e0a2%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=lXqpWrNtG5jHTcOy0QwdvcU3Xt7oKyYO9GZ1gUuCbkE%3d>.